kayenta-standalone-canary-analysis/src/main/java/com/netflix/kayenta/standalonecanaryanalysis/service/CanaryAnalysisService.java

```diff
@@ -100,10 +113,32 @@ public CanaryAnalysisExecutionResponse initiateCanaryAnalysisExecution(
   }
 
   public CanaryAnalysisExecutionStatusResponse getCanaryAnalysisExecution(
-      String canaryAnalysisExecutionId) {
-    Execution execution =
-        executionRepository.retrieve(Execution.ExecutionType.PIPELINE, canaryAnalysisExecutionId);
-    return fromExecution(execution);
+      String canaryAnalysisExecutionId, String storageAccountName) {
+
+    try {
+      Execution execution =
+          executionRepository.retrieve(Execution.ExecutionType.PIPELINE, canaryAnalysisExecutionId);
+      return fromExecution(execution);
+    } catch (ExecutionNotFoundException e) {
+      String resolvedStorageAccountName =
+          CredentialsHelper.resolveAccountByNameOrType(
+              storageAccountName,
+              AccountCredentials.Type.OBJECT_STORE,
+              accountCredentialsRepository);
+
+      StorageService storageService =
+          storageServiceRepository
+              .getOne(resolvedStorageAccountName)
+              .orElseThrow(
+                  () ->
+                      new IllegalArgumentException(
```

Comment:
this changes the behavior from before, where if the execution wasn't found `ExecutionNotFoundException` would bubble up but now it will throw `IllegalArg` - not sure if other down stream consumers depend on the type of exception thrown (or am I missing something?)
Otherwise, i guess I would expect to only try to lookup execution if a storage account is configured (again, possible missing something since I am very new to this codebase)

Reply:
similar to https://github.com/spinnaker/kayenta/pull/593/files#diff-729f4e7ba6697e7b7d60f05de492e912R35

Reply:
>I would expect to only try to lookup execution if a storage account is configured

I think this is reasonable, I will adjust the logic to clean this up.